src/metrics/metrics.ts

```diff
@@ -1,126 +1,111 @@
 import {
-  getGroupArrayInsertAtClauseForClickhouse,
-  getGroupTimeClauseForClickhouse,
-  getGroupIdClauseForClickhouse,
+  getGroupArrayInsertAtClause,
+  getGroupTimeClause,
+  getGroupIdClause,
   getInnerOrderAndLimit,
   getMergedConfig,
   getOutterOrderAndLimit,
-  getRepoWhereClauseForClickhouse,
-  getTimeRangeWhereClauseForClickhouse,
-  getUserWhereClauseForClickhouse,
-  QueryConfig
+  getRepoWhereClause,
+  getTimeRangeWhereClause,
+  getUserWhereClause,
+  QueryConfig,
+  processQueryResult,
+  getTopLevelPlatform,
+  getInnerGroupBy
 } from "./basic";
 import * as clickhouse from '../db/clickhouse';
 
 export const repoStars = async (config: QueryConfig) => {
   config = getMergedConfig(config);
   const whereClauses: string[] = ["type = 'WatchEvent'"];
-  const repoWhereClause = await getRepoWhereClauseForClickhouse(config);
+  const repoWhereClause = await getRepoWhereClause(config);
   if (repoWhereClause) whereClauses.push(repoWhereClause);
-  whereClauses.push(getTimeRangeWhereClauseForClickhouse(config));
+  whereClauses.push(getTimeRangeWhereClause(config));
 
   const sql = `
 SELECT
   id,
+  ${getTopLevelPlatform(config)},
   argMax(name, time) AS name,
-  ${getGroupArrayInsertAtClauseForClickhouse(config, { key: 'count' })}
+  ${getGroupArrayInsertAtClause(config, { key: 'count' })}
 FROM
 (
   SELECT
-    ${getGroupTimeClauseForClickhouse(config)},
-    ${getGroupIdClauseForClickhouse(config)},
+    ${getGroupTimeClause(config)},
+    ${getGroupIdClause(config)},
     COUNT() AS count
-  FROM gh_events
```

Comment:
The table name `gh_events` has been changed to `events` to indicate that the data it contains is not only from GitHub.